src/transformers/generation/utils.py

```diff
@@ -1087,6 +1087,24 @@ def _validate_model_class(self):
                 exception_message += f" Please use one of the following classes instead: {generate_compatible_classes}"
             raise TypeError(exception_message)
 
+    def _validate_assistant(self, assistant_model):
+        if assistant_model is not None:
+            if self.config.is_encoder_decoder:
+                if not assistant_model.config.is_encoder_decoder:
```

Comment:
If the assistant isn't an encoder-decoder model, then I wouldn't expect it to have `encoder_xxx` values in its config

Reply:
This is because we're only loading the decoder from an encoder-decoder model, see above comment: https://github.com/huggingface/transformers/pull/30637#discussion_r1601879851